forge-ai/src/main/java/forge/ai/ability/FogAi.java

```diff
@@ -130,7 +130,7 @@ private int countAvailableFogs(Player ai) {
 
         for (Card c : ai.getCardsIn(ZoneType.Hand)) {
             for (SpellAbility ability : c.getSpellAbilities()) {
-                if (ability.getApi().equals(ApiType.Fog)) {
+                if (ability.getApi() != null && ability.getApi().equals(ApiType.Fog)) {
```

Comment:
you could flip it `ApiType.Fog.equals(ability.getApi())`

otherwise we might look for why the API is null

Reply:
It can happen with something like Forestell special ability

So I don't think it's a bug